From 0a7f68f42f2436d119d546974f9d0c0642ffe2cc Mon Sep 17 00:00:00 2001
From: Alin Jerpelea <alin.jerpelea@sonymobile.com>
Date: Fri, 14 Aug 2015 08:45:42 +0200
Subject: [PATCH 1/2] add support for multi-arch build

for 64-bit platforms LOCAL_MODULE_RELATIVE_PATH has to be used

more info
https://source.android.com/source/64-bit-builds.html

Change-Id: I7e91f6564f262dfdbe46798c57fdc8b8fe553a66
Signed-off-by: Alin Jerpelea <alin.jerpelea@sonymobile.com>
---
 hal/Android.mk               | 2 +-
 legacy/alsa_sound/Android.mk | 6 +++---
 post_proc/Android.mk         | 2 +-
 visualizer/Android.mk        | 2 +-
 voice_processing/Android.mk  | 2 +-
 5 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/hal/Android.mk b/hal/Android.mk
index 18b8bf4..ccace92 100644
--- a/hal/Android.mk
+++ b/hal/Android.mk
@@ -56,7 +56,7 @@ endif
 
 LOCAL_MODULE := audio.primary.$(TARGET_BOARD_PLATFORM)
 
-LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
+LOCAL_MODULE_RELATIVE_PATH := hw
 
 LOCAL_MODULE_TAGS := optional
 
diff --git a/legacy/alsa_sound/Android.mk b/legacy/alsa_sound/Android.mk
index 534dd6a..3eb04a8 100644
--- a/legacy/alsa_sound/Android.mk
+++ b/legacy/alsa_sound/Android.mk
@@ -61,7 +61,7 @@ LOCAL_C_INCLUDES += system/core/include
 
 
 LOCAL_MODULE := audio.primary.msm8960
-LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
+LOCAL_MODULE_RELATIVE_PATH := hw
 LOCAL_MODULE_TAGS := optional
 
 include $(BUILD_SHARED_LIBRARY)
@@ -78,7 +78,7 @@ LOCAL_SRC_FILES := \
     AudioPolicyManagerALSA.cpp
 
 LOCAL_MODULE := audio_policy.msm8960
-LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
+LOCAL_MODULE_RELATIVE_PATH := hw
 LOCAL_MODULE_TAGS := optional
 
 LOCAL_STATIC_LIBRARIES := \
@@ -99,7 +99,7 @@ include $(CLEAR_VARS)
 
 LOCAL_PRELINK_MODULE := false
 
-LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
+LOCAL_MODULE_RELATIVE_PATH := hw
 
 LOCAL_CFLAGS := -D_POSIX_SOURCE -Wno-multichar
 LOCAL_CFLAGS += -DQCOM_ACDB_ENABLED
diff --git a/post_proc/Android.mk b/post_proc/Android.mk
index 91ed2bc..e1e3fd6 100644
--- a/post_proc/Android.mk
+++ b/post_proc/Android.mk
@@ -21,7 +21,7 @@ LOCAL_SHARED_LIBRARIES := \
 
 LOCAL_MODULE_TAGS := optional
 
-LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/soundfx
+LOCAL_MODULE_RELATIVE_PATH := soundfx
 LOCAL_MODULE:= libqcompostprocbundle
 
 LOCAL_C_INCLUDES := \
diff --git a/visualizer/Android.mk b/visualizer/Android.mk
index bec54d6..87d4987 100644
--- a/visualizer/Android.mk
+++ b/visualizer/Android.mk
@@ -27,7 +27,7 @@ LOCAL_SHARED_LIBRARIES := \
 	libdl \
 	libtinyalsa
 
-LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/soundfx
+LOCAL_MODULE_RELATIVE_PATH := soundfx
 LOCAL_MODULE:= libqcomvisualizer
 
 LOCAL_C_INCLUDES := \
diff --git a/voice_processing/Android.mk b/voice_processing/Android.mk
index b64c0e3..9b86eaf 100644
--- a/voice_processing/Android.mk
+++ b/voice_processing/Android.mk
@@ -5,7 +5,7 @@ include $(CLEAR_VARS)
 
 LOCAL_MODULE:= libqcomvoiceprocessing
 LOCAL_MODULE_TAGS := optional
-LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/soundfx
+LOCAL_MODULE_RELATIVE_PATH := soundfx
 
 LOCAL_SRC_FILES:= \
     voice_processing.c
-- 
2.8.4 (Apple Git-73)

